### DISCUSSION TITLE
Next.js - can Tailwind styles be rendered on server?

### DISCUSSION THREAD

**Discussion by cibulka** [tailwindlabs/tailwindcss#5027] [Help]
Hi! I've just started with Tailwind and I have to say I like it quite a lot.

However today I've noticed that it's classes are probably not rendered on server with Next.js. Is this expected behaviour?

I've created a detailed [issue on Next.js repo](https://github.com/vercel/next.js/issues/27360), also a demo:
* link: https://next-css-ssr-test.vercel.app/
* repo: https://github.com/cibulka/next-css-ssr-test

Am I missing something or is this expeceted behaviour? Thanks in advance!

**Comment by simonswiss** [comment]
Hey!

At the end of the day, Tailwind CSS is _Just CSS ™_. The CSS is compiled at build time, there is no JavaScript runtime involved or anything. 

So Tailwind CSS should **definitely** be rendered on the server.

I cannot see any FOUC in your example. I can see that the request preview in the network tab is unstyled. My assumption was that unless the styles are inlined in the document head (many sites do that for critical CSS), the preview in the network tab will not show the styles.

But I think I've seen some sites that load CSS via a link tag, and still show a styled network preview, as you mentioned. I honestly not sure why, and find it quite interesting.

I can guarantee you though that the CSS is loaded on the server. Tailwind is just normal CSS, by the time it makes it to the browser. And NextJS loads it via a `link` tag which is present in the server render.
